### PR TITLE
Change format of gobject, store/transmit vchData instead of hex-encoded string of a string

### PR DESCRIPTION
**Pull request by UdjinM6**
It really makes no sense to store/pass around hex string of a string... This should shrink data size to half its size as of now.

### CODE REVIEW

File: src/governance-validators.cpp [hunk 1]
```diff
@@ -22,7 +22,7 @@ CProposalValidator::CProposalValidator(const std::string& strDataHexIn)
 
 void CProposalValidator::Clear()
 {
-    strData = std::string();
+    strDataHex = std::string();
```
Comment (codablock) [inline src/governance-validators.cpp]: Just a nit: There is `.clear()` on std::string

File: src/governance-validators.cpp [hunk 2]
```diff
@@ -31,7 +31,7 @@ void CProposalValidator::Clear()
 void CProposalValidator::SetHexData(const std::string& strDataHexIn)
 {
     std::vector<unsigned char> v = ParseHex(strDataHexIn);
-    strData = std::string(v.begin(), v.end());
+    strDataHex = std::string(v.begin(), v.end());
```
Comment (codablock) [inline src/governance-validators.cpp]: Just a nit: There is `.assign(begin, end)` on std::string

**Comment by codablock** [comment]
Hmm, no protobump? Do you plan to first upgrade all nodes and then release a new version with a protobump?

**Comment by UdjinM6** [comment]
No protobump is needed, it's backwards compatible on p2p level just like #1933 

Missed a thing in serialization, fab47d0 should fix it.

**Comment by codablock** [comment]
I see that it's backwards compatible, but as we keep the protoversion on 70208, the new code/signing will never trigger until we release a version with protoversion >= 70209.

But now that I'm thinking about it it starts to make sense. If we protobump now, old nodes will immediately start to ban new nodes.

**Comment by codablock** [comment]
re-utACK https://github.com/dashpay/dash/commit/fab47d0d01a6ab077d4a797bfdf90ad00900f055

**Comment by UdjinM6** [comment]
Yes, I mean, it's implemented in such a way that we talk to old nodes in old format and to new nodes in new format i.e. it always serializes/deserializes correctly on a per node basis, even though it does it differently for different versions.

**Pull request merged**